libethash/io_posix.c

```diff
@@ -86,9 +90,22 @@ bool ethash_file_size(FILE* f, size_t* ret_size)
 
 bool ethash_get_default_dirname(char* strbuf, size_t buffsize)
 {
+#if defined(__APPLE__)
+	static const char dir_suffix[] = "Ethash/";
+	strbuf[0] = '\0';
+	char* home_dir = getenv("HOME");
+	if (!home_dir || strlen(home_dir) == 0)
```

Comment:
Not using getpwuid() simply breaks on macs. Windows isn't POSIX anyways/either, so mac is getting an exception here too. Now that you mention it, I'm going to change this back to .ethash for now. For macos, similar to windows, we have to make a special call which returns the appropriate "Application Data" folder.